llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3022,8 +3022,18 @@ LegalizerHelper::widenScalar(MachineInstr &MI, unsigned TypeIdx, LLT WideTy) {
       return UnableToLegalize;
 
     LLT Ty = MRI.getType(MI.getOperand(0).getReg());
-    if (!Ty.isScalar())
-      return UnableToLegalize;
+    if (!Ty.isScalar()) {
```

Comment:
This is going to break on the pointer case?

Reply:
I can put an assert, but it shouldn't be valid to widenScalarOrElt on a pointer vector?